pymc_marketing/mmm/linear_trend.py

```diff
@@ -0,0 +1,380 @@
+from pymc_marketing.prior import Prior, create_dim_handler
+
+
+class LinearTrend:
+    r"""LinearTrend class.
+
+    Linear trend component using change points. The trend is defined as:
+
+    .. math::
+
+        f(t) = k + \sum_{m=1}^{M} \delta_m I(t > s_m)
+
+    where:
+
+    - :math:`k` is the base trend,
+    - :math:`\delta_m` is the change in the trend at change point :math:`m`,
+    - :math:`I` is the indicator function,
+    - :math:`s_m` is the change point.
+
+    The change points are defined as:
+
+    .. math::
+
+            s_m = \frac{m}{M+1} \max(t)
+
+        where :math:`M` is the number of change points.
+
+    The priors for the trend parameters are:
+
+    - :math:`k \sim \text{Normal}(0, 0.05)`
+    - :math:`\delta_m \sim \text{Laplace}(0, 0.25)`
+
+    Adapted from MBrouns/timeseers package:
+    https://github.com/MBrouns/timeseers/blob/master/src/timeseers/linear_trend.py
+
+    Parameters
+    ----------
+    priors : dict[str, Prior], optional
+        Dictionary with the priors for the trend parameters. The
+        dictionary must have 'delta' key. If `include_intercept` is
+        True, the 'k' key is also required. By default None, or
+        the default priors.
+    dims : Dims, optional
+        Dimensions of the parameters, by default None or empty.
+    n_changepoints : int, optional
+        Number of changepoints, by default 10.
+    include_intercept : bool, optional
+        Include an intercept in the trend, by default False
+
+    Examples
+    --------
+    Linear trend with 10 changepoints:
+
+    .. code-block:: python
+
+        from pymc_marketing.mmm import LinearTrend
+
+        trend = LinearTrend(n_changepoints=10)
+
+    Use the trend in a model:
+
+    .. code-block:: python
+
+
+        import pymc as pm
+        import numpy as np
+
+        import pandas as pd
+
+        n_years = 3
+        n_dates = 52 * n_years
+        first_date = "2020-01-01"
+        dates = pd.date_range(first_date, periods=n_dates, freq="W-MON")
+        dayofyear = dates.dayofyear.to_numpy()
+        t = (dates - dates[0]).days.to_numpy()
+        t = t / 365.25
+
+        coords = {
+            "date": dates,
+        }
+        with pm.Model(coords=coords) as model:
+            intercept = pm.Normal("intercept", mu=0, sigma=1)
+            mu = intercept + trend.apply(t)
+
+            sigma = pm.Gamma("sigma", mu=0.1, sigma=0.025)
+
+            pm.Normal("obs", mu=mu, sigma=sigma, dims="date")
+
+    """
+
+    def __init__(
+        self,
+        priors: dict[str, Prior] | None = None,
+        dims: Dims | None = None,
+        n_changepoints: int = 10,
+        include_intercept: bool = False,
+    ) -> None:
+        self.n_changepoints = n_changepoints
+        self.include_intercept = include_intercept
+        self.priors: dict[str, Prior] = priors or self.default_priors.copy()
+        self.dims: Dims = dims or ()
+
+        self._checks()
+
+    def _checks(self) -> None:
```

Comment:
We could use a pydantic Base model or https://docs.pydantic.dev/latest/concepts/validation_decorator/ as in 

https://github.com/pymc-labs/pymc-marketing/blob/f2318aee2653142f39bf0465e7e6770060c731bb/pymc_marketing/mmm/mmm.py#L73-L74

Reply:
applies to all the validation steps below ;)

Reply:
Yeah, I will make that adjustment